lib/simp/rpm.rb

```diff
@@ -27,15 +28,58 @@ def self.sh(args)
     #           encountered!
     # [full_version] The full version of the package: [version]-[release]
     # [name] The full name of the package: [basename]-[full_version]
+    # [packages] An Array of package names belonging to the spec file or RPM
     def initialize(rpm_source)
       info = Simp::RPM.get_info(rpm_source)
+
       @basename = info[:name]
       @version = info[:version]
       @release = info[:release]
       @full_version = info[:full_version]
       @name = "#{@basename}-#{@full_version}"
-      @sources = Array.new
+      @arch = info[:arch]
+      @package_name = "#{@name}.#{@arch}.rpm"
       @verbose = false
+      @packages = info[:packages]
+      @signature = info[:signature]
+    end
+
+    # Returns whether or not the current RPM is newer than the passed RPM
+    def newer?(rpm)
+      return Simp::RPM.newer(@package_name, rpm) == @package_name
+    end
+
+    # Returns whether or not the first RPM is newer than the second RPM
+    def self.newer?(rpm1, rpm2)
+      return Simp::RPM.newer(rpm1, rpm2) == rpm1
+    end
+
+    # Returns the name of the newer RPM
+    #
+    # Will simply return the second RPM name if the two are equal
+    def self.newer(rpm1, rpm2)
```

Comment:
This does not work as expected.  It tells me the remote is newer when the local is newer.  For example, with  my own debugging in, got results such as the following:put
Found newer remote RPM for simp-adapter
published="simp-adapter-0.0.3-0.el7.noarch.rpm"
new=#<Simp::RPM:0x00000003e51688 @basename="simp-adapter", @version="0.0.4", @release="0.el7", @full_version="0.0.4-0.el7", @name="simp-adapter-0.0.4-0
.el7", @arch="noarch", @package_name="simp-adapter-0.0.4-0.el7.noarch.rpm", @verbose=false, @packages=["simp-adapter", "simp-adapter-pe"], @signature=n
il>